src/debug.cc

```diff
@@ -876,7 +876,8 @@ Raw::printHex(std::ostream &os) const
     const auto savedFill = os.fill('0');
     const auto savedFlags = os.flags(); // std::ios_base::fmtflags
     os << std::hex;
-    std::for_each(data_, data_ + size_,
+    auto len = (limit_ > 0 && limit_ < size_ ? limit_ : size_);
```

Comment:
Please add a printableSize() or similar method to avoid duplicating this code.

Reply:
```suggestion
```